rich/traceback.py

```diff
@@ -343,6 +393,27 @@ def safe_str(_object: Any) -> str:
             except Exception:
                 return "<exception str() failed>"
 
+        def mask_local_variable(
+            key: Any, value: Any, patterns: Optional[Iterable[str]]
+        ) -> Any:
+            """Mask locals output if key or value match regular expressions passed into `mask_locals`."""
+
+            exclude_patterns = patterns if patterns else ()
+
+            excluded_variable_name = any(
+                re.search(x, key.__repr__()) for x in exclude_patterns
```

Comment:
`__repr__` can fail. It can throw exceptions or return non-strings. When dealing with exceptions Rich as to be very defensive. See the `safe_repr` method.

Reply:
not sure how someone would take over an open PR
```
diff --git a/rich/traceback.py b/rich/traceback.py
index 3ff09c15..f84c45e4 100644
--- a/rich/traceback.py
+++ b/rich/traceback.py
@@ -4,6 +4,7 @@ import os
 import platform
 import re
 import sys
+from pprint import saferepr
 from dataclasses import dataclass, field
 from traceback import walk_tb
 from types import ModuleType, TracebackType
@@ -401,10 +402,10 @@ class Traceback:
             exclude_patterns = patterns if patterns else ()
 
             excluded_variable_name = any(
-                re.search(x, key.__repr__()) for x in exclude_patterns
+                re.search(x, saferepr(key)) for x in exclude_patterns
             )
             excluded_variable_value = any(
-                re.search(x, value.__repr__()) for x in exclude_patterns
+                re.search(x, saferepr(value)) for x in exclude_patterns
             )
 
             if excluded_variable_name or excluded_variable_value:
```